core/activity_pub.md

```diff
@@ -0,0 +1,40 @@
+
+The usefulness of this conceptual nesting is addressing the **right to individual exit** and the **right to collective exit**. 
+
+The former can be relatively easily addressed in Activity Pub by using flexible URI's such as DID's. This would give any user power to exit a community without losing it's account information (content may be lost, but followers/following lists would not).
+
+The latter requires the concept of communal ownership, which Activity Pub lacks. Suppose an Activity Pub implementation added support for DID's and thus fulfilled the requirements for the right to individual exit. In the presence of a malicious operator, we can reasonably predict that even in the case of an organized exit, the size and network effect of the forked community will be less than the original community. The right to collective exit and implicitly the right to partial ownership of a community helps mitigates the risk of malicious operators while aligning the incentives of operators, producers, moderators, and consumers.
+
+
+## Public/Private Key Authentication
+
+ActivityPub does not specify a specific authentication scheme, however most ActivityPub communities use OAuth2.0 or something similar which involves a traditional username/password scheme. Since all of the community state must be public and shareable between hosts, we cannot rely on username/password schemes or any scheme that gives hosts privileged information on a user’s authentication info. Thus, users must authenticate by proving ownership of a secret private key that corresponds to the public key that is publicly registered on a blockchain.
+
+## URI and DID
+
+URIs (Uniform Resource Identifier) are used within Activity Pub to identify different servers. 
+
+Fora will use [decentralized identifiers](https://www.w3.org/TR/did-core/), to identify both users and platforms. These DIDs must be registered on a blockchain that the platform can query identities from. Users will have a DID so they can self-attest to their identity on the blockchain and then use that DID in any Fora community they wish. They can also retain this identity even as they move to different communities. The DID document for each user must contain the current public key they will use sign messages sent from their Fora user account. 
+
+Fora implementations which rely on the public key registered with a DID instead of registering a public key with a governance module, will benefit from dynamic swapping of public keys. 
+
+Meanwhile communities will also have DIDs that identify them to their users. This DID will also be registered on the blockchain as a government. The DID will resolve to the current hosts of the platform. Any governance decisions which affects the set of hosts will update the DID resolving result to output the new set of hosts.
```

Comment:
It should be clear that a community does not have a defined set of hosts. Anyone can be a host.

What we want to say here is that a community may define a default set of hosts as a UX convenience.

```suggestion
Meanwhile communities will also have DIDs that identify them to their users. This DID will also be registered on the blockchain as a government. The DID may include a default set of "seed" hosts to allow users to conveniently find and connect to the community. Any governance decisions which affects the set of hosts will update the DID resolving result to output a new set of hosts.

Note: There may be many valid, honest hosts that exist outside of this community-curated set.
```